backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/RemoveImageCommand.java

```diff
@@ -323,4 +331,24 @@ protected VDSReturnValue performDeleteImageVdsmOperation() {
                                 getDiskImage().isWipeAfterDelete(), getStorageDomain().getDiscardAfterDelete(),
                                 getParameters().isForceDelete())));
     }
+
+    @Override
+    public CommandCallback getCallback() {
+        return getParameters().getParentCommand() == ActionType.RemoveDisk ?
+                callbackProvider.get() :
+                null;
+    }
+
+    public void onSucceeded() {
+        performImageDbOperations();
+    }
+
+    public void onFailed() {
+        imageDao.updateStatusOfImagesByImageGroupId(getImageGroupId(), ImageStatus.ILLEGAL);
```

Comment:
a question related to what you asked @bennyz and me - if the image is changed to status ILLEGAL, what happens the next time we try to remove it?

Reply:
Then it needs to be handled, but I thought we should better decide first.
The more I think about it - it seems to me that your solution with removing from DB and leaving audit log is better

Reply:
Maybe we can try to remove it without wiping if it fails with wiping?

Reply:
> The more I think about it - it seems to me that your solution with removing from DB and leaving audit log is better

yeah, unless the user can simply remove such ILLEGAL images - because the problem with the audit log is that it could be easily missed but if we set the status to ILLEGAL, the user should probably check why. however, if the user is stuck with those ILLEGAL images then yeah, let's just have an audit log (I think that when we'll try to remove the image again we won't find it and consider that as a successful removal, but I'm not sure)

> Maybe we can try to remove it without wiping if it fails with wiping?

nah, that would be problematic - the whole point of 'wipe after delete' is to ensure the data won't be accessible to anyone when the image is removed